libraries/PDM/src/PDM.cpp

```diff
@@ -48,10 +48,15 @@ int PDMClass::begin(int channels, long sampleRate)
 {
   _channels = channels;
 
+  NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
+  NRF_CLOCK->TASKS_HFCLKSTART    = 1;
+  while (NRF_CLOCK->EVENTS_HFCLKSTARTED == 0) { }
```

Comment:
Should this be disabled in `.end()`?

Reply:
I thought about this and it might not be clever (since BLE could be in use and would stop working on `PDM.end()` ).

Reply:
ok, so then should we check if it's already enabled before "restarting" it. I would also add a small comment to capture why end() doesn't disable to be kind to our future selves :)

Reply:
Agree on the comment for our future selves :wink: